### PR TITLE
fix src/AppBundle/Security/TimeAuthenticator.php

### PR DESCRIPTION
**Pull request by Reyh**
$isPasswordValid was emty
isPasswordValid attempted UserInterface first argument, string given

### CODE REVIEW

**Comment by xabbuh** [comment]
Shouldn't we remove the `if` below instead? I fail to see how `$isPasswordValid` could ever be `false`.

**Comment by Reyh** [comment]
```
        if ($currentUser instanceof UserInterface) {
            if ($currentUser->getPassword() !== $user->getPassword()) {
                throw new BadCredentialsException('The credentials were changed from another session.');
            }
        } else {
            if ('' === ($givenPassword = $token->getCredentials())) {
                throw new BadCredentialsException('The given password cannot be empty.');
            }
            if (!$this->encoder->isPasswordValid($user->getPassword(), $givenPassword, $user->getSalt())) {
                throw new BadCredentialsException('The given password is invalid.');
            }
        }
        if ($isPasswordValid) {
                ...
        }
```

I think $isPasswordValid was undefined

**Comment by xabbuh** [comment]
That's right. But what I mean is that even if we initialise the variable now, it will never be `false` when the `if ($isPasswordValid)` statement is reached (because in those cases the code above would have already exited the method with an exception), right?

File: security/custom_password_authenticator.rst [hunk 1]
```diff
@@ -54,16 +54,16 @@ the user::
...
-                if ($currentUser->getPassword() !== $user->getPassword()) {
+                if ($isPasswordValid = $currentUser->getPassword() !== $user->getPassword()) {
                     throw new BadCredentialsException('The credentials were changed from another session.');
                 }
             } else {
                 if ('' === ($givenPassword = $token->getCredentials())) {
                     throw new BadCredentialsException('The given password cannot be empty.');
                 }
-                if (!$this->encoder->isPasswordValid($user->getPassword(), $givenPassword, $user->getSalt())) {
+                if (!$isPasswordValid=$this->encoder->isPasswordValid($user, $givenPassword, $user->getSalt())) {
```
Comment (apfelbox) [inline security/custom_password_authenticator.rst]: This is wrong, should be:

```php
if ($isPasswordValid = !$this->encoder->isPasswordValid($user, $givenPassword, $user->getSalt())) {
```

(look at the position of `!`)

**Comment by apfelbox** [comment]
I agree with @xabbuh, we should just remove the `if`.

**Comment by weaverryan** [comment]
Wait a second. This whole new code block is a bit insane :). I see it was added in #10100 to fix #4579. 

Let's fix any logic needed here. But this article should be re-written to support Guard. 

Do we know what we want the logic to look like exactly?

**Comment by javiereguiluz** [comment]
@Reyh thanks a lot for this contribution and for helping us improve the current code. Sadly this pull request "stalled" and there are merge conflicts, so I've opened #10448 to make the alternative fix proposed by @xabbuh. In any case, we hope to see more contributions from you in the future for Symfony Docs. Thanks a lot!

**Pull request closed without merging**